### PR TITLE
BigQuery selective partition load on merge

### PR DESCRIPTION
**Pull request by hayssams**
## Summary
In the MergeOptions queryFIlter, the user may now write something like this : `partitionField in last(3)`
this will be translated to `partitionField between partitionStart and partitionEnd`

partitionEnd is the last partition in the dataset
paritionStart is the 3rd last partition in the dataset

if partititionStart or partitionEnd does nos exist (aka empty dataset) they are replaced by 19700101
**PR Type: Feature**

**Status: Ready to review**

**Breaking change? No**

### CODE REVIEW

File: src/main/scala/com/ebiznext/comet/workflow/IngestionWorkflow.scala [hunk 1]
```diff
@@ -291,7 +291,7 @@ class IngestionWorkflow(
     schemas.partition { case (schema, _) => schema.isDefined }
   }
 
-  private[this] def predicate(domain: Domain, schemasName: List[String], file: Path): Boolean = {
+  private def predicate(domain: Domain, schemasName: List[String], file: Path): Boolean = {
```
Comment (mhdkassir) [inline src/main/scala/com/ebiznext/comet/workflow/IngestionWorkflow.scala:294]: is this a Codacy recommandation ?

File: src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala [hunk 2]
```diff
@@ -883,29 +939,30 @@ trait IngestionJob extends SparkJob {
           .fields
           .length
       )
-        mergeParquet(withScriptFieldsDF, existingDF, mergeOptions)
+        applyMergeParquet(withScriptFieldsDF, existingDF, mergeOptions)
       else
         throw new RuntimeException(
           "Input Dataset and existing HDFS dataset do not have the same number of columns. Check for changes in the dataset schema ?"
         )
-      mergeParquet(withScriptFieldsDF, existingDF, mergeOptions)
```
Comment (mhdkassir) [inline src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala:891]: mergeParquet was called twice before ?

Comment (seyguai) [reply src/main/scala/com/ebiznext/comet/job/ingest/IngestionJob.scala:891]: Yes

**Review by mhdkassir — APPROVED** [review]
LGTM

File: src/main/scala/com/ebiznext/comet/workflow/IngestionWorkflow.scala (hunk 1 above)
Comment (hayssams) [reply src/main/scala/com/ebiznext/comet/workflow/IngestionWorkflow.scala:294]: yes

**Pull request merged**